src/Domain.LinnApps/Reports/ShortagesReportService.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.Reports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Common.Reporting.Models;
+    using Linn.Purchasing.Domain.LinnApps.Reports.Models;
+
+    public class ShortagesReportService : IShortagesReportService
+    {
+        private readonly IRepository<ShortagesEntry, string> shortagesEntryView;
+
+        public ShortagesReportService(
+            IRepository<ShortagesEntry, string> shortagesEntryView)
+        {
+            this.shortagesEntryView = shortagesEntryView;
+        }
+
+        public IEnumerable<ResultsModel> GetReport(
+            int purchaseLevel,
+            string vendorManager)
+        {
+            
+            var results = this.shortagesEntryView.FilterBy(x => (Convert.ToInt32(x.PurchaseLevel) <= purchaseLevel) &&
+                                                                            (vendorManager == "ALL" || vendorManager == x.VendorManagerCode)).ToList();
+
+            var returnResults = new List<ResultsModel>();
+
+            foreach (var shortagesForPlanner in results.GroupBy(a => new { a.PlannerName }))
+            {
+                var model = new ResultsModel();
+
+                model.AddColumn("VendorManagerCode", "Vendor Manager Code");
+                model.AddColumn("VendorManagerName", "Vendor Manager Name");
+
+                model.ReportTitle = new NameModel(
+                    $"Purchasing shortages planner");
+                model.RowHeader = "Planner";
+
+                var distinctVendorManagers = shortagesForPlanner.DistinctBy(x => x.VendorManagerCode);
+                var distinctLevels = shortagesForPlanner.DistinctBy(x => x.PurchaseLevel);
+
+                foreach (var level in distinctLevels)
```

Comment:
you should probably order this list before looping throught it and adding columns (Otherwise your columns could be in a random order?

Reply:
Was going to ask about this. Will add in.